### ISSUE TITLE
Stuck updating files

### ISSUE THREAD

**Issue by igorkulman** [mswgen/oc-updater#28]
Trying to update my OpenCore 0.8.6 on macOS 12.7.2 I just always get the app to be stuck in this step

<img width="490" alt="Screenshot 2024-03-23 at 11 28 05" src="https://github.com/mswgen/oc-updater/assets/4746070/e9cdfe22-0951-4db8-90b8-068bde330af7">

**Comment by dropdb220** [comment]
Sorry for the late reply. (I entered high school last month and I'm very busy now.)
Is the issue still present in the latest version(1.4.2)?

**Comment by dropdb220** [comment]
@igorkulman

**Comment by igorkulman** [comment]
Testing now with 1.4.2 and I am still stuck.

**Comment by dropdb220** [comment]
Please run it from terminal by executing `/Applications/OpenCore\ Updater.app/Contents/MacOS/OpenCore\ Updater` and provide the console output.

**Comment by igorkulman** [comment]
```
❯ /Applications/OpenCore\ Updater.app/Contents/MacOS/OpenCore\ Updater
Checking for update
[1099:0602/102409.812111:ERROR:trust_store_mac.cc(750)] Error parsing certificate:
ERROR: Failed parsing extensions

Update for version 1.4.2 is not available (latest version: 1.4.2, downgrade is disallowed).
checkForUpdatesAndNotify called, downloadPromise is null
(node:1099) UnhandledPromiseRejectionWarning: Error: Command failed: cp -r /Users/igorkulman/.oc-update/676171/VirtualSMC/Kexts/VirtualSMC.kext /Users/igorkulman/Desktop/EFI 2/OC/Kexts
cp: 2/OC/Kexts is not a directory

    at ChildProcess.exithandler (node:child_process:422:12)
    at ChildProcess.emit (node:events:517:28)
    at maybeClose (node:internal/child_process:1098:16)
    at Socket.<anonymous> (node:internal/child_process:450:11)
    at Socket.emit (node:events:517:28)
    at Pipe.<anonymous> (node:net:350:12)
(Use `OpenCore Updater --trace-warnings ...` to show where the warning was created)
(node:1099) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
```

**Comment by dropdb220** [comment]
The cause of this error is that the path to EFI folder contains a space and the program does not handle it correctly.
Will fix by using the native fs module instead of relying on shell scripts.

**Comment by dropdb220** [comment]
Resolved in 1.4.3
